py/objstr.c

```diff
@@ -832,6 +832,10 @@ STATIC bool arg_looks_numeric(mp_obj_t arg) {
     ;
 }
 
+STATIC bool arg_looks_string(mp_obj_t arg) {
+    return MP_OBJ_IS_TYPE(arg, &mp_type_str) || MP_OBJ_IS_STR(arg);
```

Comment:
Did you see definition of MP_OBJ_IS_STR()? I'm skipping further questions like "Why there's a function for once-off use?" or "Why it's called 'looks like string', if actually it checks that something *is string*?"